pkg/managed/secrets_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package managed
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"github.com/verrazzano/verrazzano-operator/pkg/constants"
+	"github.com/verrazzano/verrazzano-operator/pkg/monitoring"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutil"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+// SecretProperties contains the set of properties used to identify a secret during testing
+type SecretProperties struct {
+	name      string
+	namespace string
+}
+
+// TestCreateSecretes tests creation of secrets
+// GIVEN a model binding pair and cluster connections configured with required secrets
+//  WHEN the CreateSecrets is called
+//  THEN the required secrets are generated and associated to the specified namespaces
+func TestCreateSecrets(t *testing.T) {
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+
+	vmiSecret := monitoring.NewVmiSecret(modelBindingPair.Binding)
+	secrets := &testutil.FakeSecrets{Secrets: map[string]*corev1.Secret{
+		constants.VmiSecretName: vmiSecret,
+	}}
+
+	CreateSecrets(modelBindingPair, clusterConnections, clusterConnection.KubeClient, secrets)
+	assert := assert.New(t)
```

Comment:
minor: if this causes an IDE naming conflict warning, you can get around it by naming the import to something like ```asserts "github.com/stretchr/testify/assert"```

Reply:
My IDE isn't indicating an issue...